test/verify/check-networking-firewall

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python2
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import parent
+from testlib import *
+import time
+
+
+def wait_unit_state(machine, unit, state, timeout=10):
+    start = time.time()
+    while machine.execute("systemctl is-active {} || true".format(unit)).strip() != state:
+        time.sleep(0.1)
+        if time.time() - start > timeout:
+            raise Exception("timed out waiting for {} to become {}".format(unit, state))
+
+
+class TestNetworking(MachineCase):
+
+    def testNetworkingPage(self):
+        b = self.browser
+        m = self.machine
+
+        self.login_and_go("/network")
+        b.wait_visible("#networking-firewall")
+
+        m.execute("systemctl stop firewalld")
+        b.wait_in_text("#networking-firewall-switch label.active", "Off")
+        m.execute("systemctl start firewalld")
+        b.wait_in_text("#networking-firewall-switch label.active", "On")
+
+        # to toggle the switch, click on the non-active label, where the slider is
+        b.click("#networking-firewall-switch label:not(.active)")
+        b.wait_present("#networking-firewall-switch label.active:not(.disabled)")
+        b.wait_in_text("#networking-firewall-switch label.active", "Off")
+        wait_unit_state(m, "firewalld", "inactive")
+        b.wait_in_text("#networking-firewall-summary", "0 Active Rules")
+
+        b.click("#networking-firewall-switch label:not(.active)")
+        b.wait_present("#networking-firewall-switch label.active:not(.disabled)")
+        b.wait_in_text("#networking-firewall-switch label.active", "On")
+        wait_unit_state(m, "firewalld", "active")
+
+        active_rules = m.execute("firewallctl zone '' list services").split()
+        b.wait_in_text("#networking-firewall-summary", "{} Active Rules".format(len(active_rules)))
```

Comment:
Corner case: Use "Rule" here instead of "Rules", for the corner case that there is really just one. However, our images should have at least ssh and cockpit, so I guess it's fine.

Reply:
Wow, awesome catch!